modules/fedimint-walletv2-client/src/lib.rs

```diff
@@ -0,0 +1,570 @@
+            .make_client_outputs(ClientOutputBundle::new_no_sm(vec![client_output]));
+
+        self.client_ctx
+            .finalize_and_submit_transaction(
+                operation_id,
+                WalletCommonInit::KIND.as_str(),
+                |txid, change_outpoints| {
+                    WalletOperationMeta::Send(SendMeta {
+                        txid,
+                        change_outpoints,
+                        address: address.clone(),
+                        amount,
+                        fee: send_fee.value,
+                    })
+                },
+                TransactionBuilder::new().with_outputs(client_output_bundle),
+            )
+            .await
+            .map_err(|_| SendError::InsufficientFunds)?;
+
+        Ok(operation_id)
+    }
+
+    /// Await the final operation state of a send or receive operation.
+    pub async fn await_final_operation_state(
+        &self,
+        operation_id: OperationId,
+    ) -> FinalOperationState {
+        match self
+            .client_ctx
+            .transaction_updates(operation_id)
+            .await
+            .await_any_tx_accepted()
+            .await
+        {
+            Ok(()) => FinalOperationState::Success,
+            Err(..) => FinalOperationState::Aborted,
+        }
+    }
+
+    /// Increment the address counter and return the new highest index.
+    pub async fn increment_address_index(&self) -> u64 {
+        let mut dbtx = self.db.begin_transaction().await;
+
+        let index = dbtx.get_value(&AddressCounterKey).await.unwrap_or(0);
+
+        dbtx.insert_entry(&AddressCounterKey, &(index + 1)).await;
+
+        dbtx.commit_tx().await;
+
+        index
+    }
+
+    /// Return the number of all previously derived addresses.
+    pub async fn address_count(&self) -> u64 {
+        self.db
+            .begin_transaction()
+            .await
+            .get_value(&AddressCounterKey)
+            .await
+            .unwrap_or(0)
+    }
+
+    /// Derive address for an index.
+    pub fn derive_address(&self, index: u64) -> Address {
+        descriptor(
+            &self.cfg.bitcoin_pks,
+            &self.derive_tweak(index).public_key().consensus_hash(),
+        )
+        .address(self.cfg.network)
+    }
+
+    fn derive_tweak(&self, index: u64) -> Keypair {
+        self.root_secret
+            .child_key(ChildId(index))
+            .to_secp_key(secp256k1::SECP256K1)
+    }
+
+    fn default_esplora_server(&self) -> Option<SafeUrl> {
+        let url = match self.cfg.network {
+            Network::Bitcoin => "https://blockstream.info/api/",
+            Network::Testnet => "https://blockstream.info/testnet/api/",
+            Network::Signet => "https://blockstream.info/signet/api/",
+            _ => return None,
+        };
+
+        Some(SafeUrl::parse(url).expect("Failed to parse default esplora server"))
+    }
+
+    /// Check an address for unspent deposits and return the deposits in
```

Comment:
The format of a docstring comment should be a "single sentence, newlines, longer description". I thought clippy nags if it isn't the case.